fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/ResourceConditions.java

```diff
@@ -0,0 +1,176 @@
+import java.util.function.Predicate;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.resource.JsonDataLoader;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.JsonHelper;
+
+import net.fabricmc.fabric.impl.resource.conditions.ResourceConditionsImpl;
+
+/**
+ * Registration and access to resource loading conditions.
+ * A resource condition is an identified {@code Predicate<JsonObject>} that can decide whether a resource should be loaded or not.
+ * <ul>
+ *     <li>A JSON object that may contain a condition can be parsed with {@link #objectMatchesConditions}.
+ *     This is the preferred way of implementing conditional objects, as it handles the details of the format (see below) and catches and logs thrown exceptions.</li>
+ *     <li>The lower-level {@link #conditionsMatch} and {@link #conditionMatches} may be useful when implementing conditions.</li>
+ *     <li>Conditions are registered with {@link #register} and queried with {@link #get}.</li>
+ * </ul>
+ *
+ * <p>At the moment, Fabric only recognizes conditions for resources loaded by subclasses of {@link JsonDataLoader}.
+ * This means: recipes, advancements, loot tables, loot functions and loot conditions.
+ *
+ * <p>Fabric provides some conditions, which can be generated using the helper methods in {@link DefaultResourceConditions}.
+ *
+ * <h3>Details of the format</h3>
+ *
+ * <p>A conditional JSON object must have a {@link #CONDITIONS_KEY} entry, containing an array of condition objects.
+ * The conditions in the array must be satisfied to load the resource.
+ * Each condition object must contain a {@link #CONDITION_ID} entry with the identifier of the condition,
+ * and it may also contain additional data for the condition.
+ * Here is an example of a resource that is only loaded if no mod with id {@code a} is loaded:
+ * <pre>{@code
+ * {
+ *   ... // normal contents of the resource
+ *   "fabric:load_conditions": [ // array of condition objects
+ *     { // a condition object
+ *       // the identifier of the condition... the "fabric:not" condition inverts the condition in its "value" field
+ *       "condition": "fabric:not",
+ *       // additional data, for "fabric:not", the "value" field is required to be another condition object
+ *       "value": {
+ *         // the identifier of the condition
+ *         "condition": "fabric:all_mods_loaded",
+ *         // additional data, for "fabric:all_mods_loaded"
+ *         "values": [
+ *           "a"
+ *         ]
+ *       }
+ *     }
+ *   ]
+ * }
+ * }</pre>
+ */
+public final class ResourceConditions {
+	private static final Map<Identifier, Predicate<JsonObject>> REGISTERED_CONDITIONS = new ConcurrentHashMap<>();
+	/**
+	 * The key fabric uses to identify resource conditions in a JSON object.
```

Comment:
```suggestion
	 * The key ({@value}) Fabric uses to identify resource conditions in a JSON object.
```
This shows the actual key as well in the generated javadoc (without going to the list of constants).

Reply:
Interesting!